CFC_DataCollector/trip_generator/trip_gen.py

```diff
@@ -0,0 +1,136 @@
+from recommender.otp import OTP, PathNotFoundException
+from pygeocoder import Geocoder
+import random, math
+from recommender.trip import Coordinate 
+import json
+import datetime
+
+class Address:
+
+    ## This class exisists only for caching purposes
+    ## So we don't have to call google maps a million times
+
+    def __init__(self, address):
+        self.text = address
+        self.cord = None
+    
+    def __str__(self):
+        return self.text            
+
+class Creator: 
+
+    def __init__(self):
+        self.starting_points = [ ]
+        self.ending_points = [ ]
+        self.a_to_b = set( )
+        self.num_trips = None
+        self.radius = None
+        self.amount_missed = 0
+
+    def get_starting_ending_points(self):
+        city_file = open("trip_generator/input.json", "r") ## User (Naomi) specifies locations and radius they want
+        jsn = json.load(city_file)
+        self.num_trips = jsn["number of trips"]
+        self.radius = int(jsn["radius"])
+        for address in jsn["starting centroids"]:
+            self.starting_points.extend(generate_random_locations_in_radius(Address(address), self.radius, self.num_trips/len(jsn["starting centroids"])))
+        for address in jsn["ending centroids"]:
+            self.ending_points.extend(generate_random_locations_in_radius(Address(address), self.radius, self.num_trips/len(jsn["starting centroids"])))
+     
+    def make_a_to_b(self):
+        for _ in xrange(self.num_trips + self.radius*3):  ## Based on very rough estimate of how many of these end up in the ocean
+            start_index = random.randint(0, len(self.starting_points) - 1)
+            end_index = random.randint(0, len(self.ending_points) - 1)
+            starting_point = self.starting_points[start_index]
+            ending_point = self.ending_points[end_index]
+            to_add = ( starting_point, ending_point )
+            self.a_to_b.add(to_add)
+
+
+    def get_trips_from_a_to_b(self):
+        mode_tuple = ("CAR", "WALK", "BICYCLE", "TRANSIT")
+        curr_time = datetime.datetime.now()
+        curr_month = curr_time.month
+        curr_year = curr_time.year
+        curr_day = curr_time.day
+        curr_hour = curr_time.hour
+        curr_minute = curr_time.minute
+        for t in self.a_to_b:
+            mode = mode_tuple[random.randint(0, len(mode_tuple) - 1)] ## Unsophisticated mode choice
```

Comment:
Enhancement: Allow people to specify the random seed so that we can get arbitrary, but reproducible trips.